neuraxle/pipeline.py

```diff
@@ -369,99 +277,56 @@ class ResumablePipeline(Pipeline, ResumableStepMixin):
     Fits and transform steps after latest checkpoint
     """
 
-    def __init__(self, steps: NamedTupleList, pipeline_saver: PipelineSaver = None):
-        Pipeline.__init__(self, steps=steps)
-
-        if pipeline_saver is None:
-            self.pipeline_saver = JoblibPipelineSaver(DEFAULT_CACHE_FOLDER)
-        else:
-            self.pipeline_saver = pipeline_saver
-
-    def save(self, data_container: DataContainer):
-        """
-        Save the fitted parent pipeline with the passed data container
-
-        :param data_container: data container to save pipeline with
-        :return:
-        """
-        if self.parent_step is None:
-            self.pipeline_saver.save(self, data_container)
-        else:
-            self.parent_step.save(data_container)
-
-    def _load_checkpoint(self, data_container: DataContainer) -> Tuple[NamedTupleList, DataContainer]:
+    def _load_checkpoint(
+            self,
+            data_container: DataContainer,
+            context: ExecutionContext
+    ) -> Tuple[NamedTupleList, DataContainer]:
         """
         Try loading a pipeline cache with the passed data container.
         If pipeline cache loading succeeds, find steps left to do,
         and load the latest data container.
 
         :param data_container: the data container to resume
+        :param context: the execution context to resume
         :return: tuple(steps left to do, last checkpoint data container)
         """
         new_starting_step_index, starting_step_data_container = \
-            self._get_starting_step_info(data_container)
-
-        if self.parent_step is None:
-            if not self.pipeline_saver.can_load(self, starting_step_data_container):
-                return self.steps_as_tuple, data_container
+            self._get_starting_step_info(data_container, context)
 
-            saved_pipeline = self.pipeline_saver.load(self, starting_step_data_container)
+        loaded_pipeline = self.load(context)
```

Comment:
Good to see this method reduced in size. This is a sign we did something good. The pipeline class was too big and had too many responsibilities before which was a symptom of OOP design problems that we now have finally fixed.